agent/internal/options.go

```diff
@@ -81,9 +81,10 @@ type SecurityOptions struct {
 
 // TLSOptions is the TLS connection configuration for the agent.
 type TLSOptions struct {
-	Enabled    bool   `json:"enabled"`
-	SkipVerify bool   `json:"skip_verify"`
-	MasterCert string `json:"master_cert"`
+	Enabled        bool   `json:"enabled"`
+	SkipVerify     bool   `json:"skip_verify"`
+	MasterCert     string `json:"master_cert"`
+	MasterCertAddr string `json:"master_cert_addr"`
```

Comment:
Rename this to be "MasterServerName" or something.  Calling it an "address" is not super clear, since hostnames on certs are not normally allowed to be ip addresses at all.